Index: rosdep.yaml
===================================================================
--- rosdep.yaml	(revision 140)
+++ rosdep.yaml	(working copy)
@@ -3,28 +3,48 @@
   debian: libtbb-dev
   fedora: tbb
   macports: tbb
+  osxbrew:
+    homebrew:
+      packages: [ tbb ]
 libvtk:
   ubuntu: libvtk5-dev
   debian: libvtk5-dev
   fedora: vtk-devel
   macports: vtk-devel
+  osxbrew:
+    homebrew:
+      packages: [ vtk ]
+      args: [ --python ]
 unzip:
   ubuntu: unzip
   debian: unzip
   fedora: unzip
   macports: unzip
+  osxbrew:
+    homebrew:
+      packages: |
 hdf5:
   ubuntu: libhdf5-serial-dev
   debian: libhdf5-serial-dev
   fedora: hdf5-devel
   macports: hdf5-18
+  osxbrew:
+    homebrew:
+      packages: [ hdf5 ]
 libqhull:
   ubuntu: libqhull-dev
   debian: libqhull-dev
   fedora: qhull-devel
   macports: qhull
+  osxbrew:
+    homebrew:
+      packages: [ qhull ]
 cmake:
   ubuntu: cmake
   debian: cmake
   fedora: cmake
   macports: cmake
+  osxbrew:
+    homebrew:
+      packages: [ cmake ]
+
Index: pcl/osx_interactor_workaround.patch
===================================================================
--- pcl/osx_interactor_workaround.patch	(revision 0)
+++ pcl/osx_interactor_workaround.patch	(revision 0)
@@ -0,0 +1,31 @@
+--- visualization/src/interactor.cpp	2011-08-31 15:56:15.000000000 -0500
++++ visualization/src/interactor.cpp	2011-10-19 14:46:33.000000000 -0500
+@@ -55,14 +55,16 @@
+     void 
+     PCLVisualizerInteractor::stopLoop ()
+     {
+-#if defined _WIN32 || defined VTK_USE_COCOA || defined VTK_USE_CARBON
++#if defined _WIN32
+       BreakLoopFlagOn ();
+       // Send a VTK_BreakWin32Loop ClientMessage event to be sure we pop out of the
+       // event loop.  This "wakes up" the event loop.  Otherwise, it might sit idle
+       // waiting for an event before realizing an exit was requested.
+-#if defined _WIN32
++#  if defined _WIN32
+       SendMessage (this->WindowId, RegisterWindowMessage (TEXT ("VTK_BreakWin32Loop")), 0, 0);
+-#endif
++#  endif
++#elif defined VTK_USE_COCOA || defined VTK_USE_CARBON
++      // Do Nothing for now
+ #else
+       BreakLoopFlagOn ();
+       XClientMessageEvent client;
+@@ -77,7 +79,7 @@
+ #endif
+     }
+ 
+-#if defined _WIN32 || defined VTK_USE_COCOA || defined VTK_USE_CARBON
++#if defined _WIN32
+     //////////////////////////////////////////////////////////////////////////
+     void 
+     PCLVisualizerInteractor::Start ()
Index: pcl/Makefile
===================================================================
--- pcl/Makefile	(revision 140)
+++ pcl/Makefile	(working copy)
@@ -4,6 +4,7 @@
 TARBALL_URL = https://code.ros.org/svn/release/download/thirdparty/pcl/pcl-1.1.1.tar.gz
 SOURCE_DIR = build/pcl
 UNPACK_CMD = tar xzf
+TARBALL_PATCH = osx_interactor_workaround.patch
 include $(shell rospack find mk)/download_unpack_build.mk
 
 #SVN_DIR = build/pcl_trunk
@@ -22,6 +23,7 @@
 	export LD_LIBRARY_PATH=`rospack find std_msgs`/lib:`rospack find sensor_msgs`/lib:`rospack find roscpp_serialization`/lib:`rospack find cpp_common`/lib:`rospack find rostime`/lib:`rospack find roscpp_traits`/lib:`rospack find roscpp`/lib:`rospack find rosconsole`/lib:`rospack find std_msgs`/lib:`rospack find sensor_msgs`/lib:$$LD_LIBRARY_PATH && \
 	cmake -DCMAKE_INSTALL_PREFIX=`pwd`/../../.. \
 				-DCMAKE_BUILD_TYPE=Release \
+				-DCMAKE_OSX_ARCHITECTURES="x86_64" \
 				-DUSE_ROS=ON \
           -DFLANN_INCLUDE_DIR=`rospack find flann`/include \
           -DFLANN_LIBRARY=`rospack find flann`/lib/libflann.so \
