diff -r 9d58710fc7db collada_urdf/CMakeLists.txt
--- a/collada_urdf/CMakeLists.txt	Tue Sep 06 11:51:06 2011 -0700
+++ b/collada_urdf/CMakeLists.txt	Wed Oct 19 19:45:42 2011 -0500
@@ -4,6 +4,10 @@
 rosbuild_init()
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 
+if(APPLE)
+    link_libraries(tinyxml)
+endif(APPLE)
+
 rosbuild_add_library(collada_urdf src/collada_urdf.cpp)
 rosbuild_link_boost(collada_urdf system)
 
diff -r 9d58710fc7db kdl_parser/CMakeLists.txt
--- a/kdl_parser/CMakeLists.txt	Tue Sep 06 11:51:06 2011 -0700
+++ b/kdl_parser/CMakeLists.txt	Wed Oct 19 19:45:42 2011 -0500
@@ -23,6 +23,10 @@
 #uncomment if you have defined services
 #rosbuild_gensrv()
 
+if(APPLE)
+    link_libraries(tinyxml)
+endif(APPLE)
+
 rosbuild_add_library(${PROJECT_NAME} src/kdl_parser.cpp)
 
 rosbuild_add_executable(check_kdl_parser src/check_kdl_parser.cpp )
diff -r 9d58710fc7db rosdep.yaml
--- a/rosdep.yaml	Tue Sep 06 11:51:06 2011 -0700
+++ b/rosdep.yaml	Wed Oct 19 19:45:42 2011 -0500
@@ -2,3 +2,7 @@
   ubuntu: libpcre3-dev
   fedora: pcre-devel
   debian: libpcre3-dev
+  osxbrew:
+    homebrew:
+      packages: [ pcre++]
+
