Index: wxswig/Makefile
===================================================================
--- wxswig/Makefile	(revision 14716)
+++ wxswig/Makefile	(working copy)
@@ -14,9 +14,9 @@
 
 wxswig: $(SOURCE_DIR)/unpacked
 ifneq ($(MAKE),)
-	cd $(SOURCE_DIR) && ./configure --prefix=$(shell rospack find wxswig) && $(MAKE) $(ROS_PARALLEL_JOBS) && $(MAKE) install
+	cd $(SOURCE_DIR) && ./configure --prefix=$(shell rospack find wxswig) --disable-dependency-tracking && $(MAKE) $(ROS_PARALLEL_JOBS) && $(MAKE) install
 else
-	cd $(SOURCE_DIR) && ./configure --prefix=$(shell rospack find wxswig) && make $(ROS_PARALLEL_JOBS) && make install
+	cd $(SOURCE_DIR) && ./configure --prefix=$(shell rospack find wxswig) --disable-dependency-tracking && make $(ROS_PARALLEL_JOBS) && make install
 endif
 	mv bin/swig bin/swig-real
 	cp swig-wrap bin/swig
Index: rxtools/src/rxtools/rosout_list_control.cpp
===================================================================
--- rxtools/src/rxtools/rosout_list_control.cpp	(revision 14716)
+++ rxtools/src/rxtools/rosout_list_control.cpp	(working copy)
@@ -732,8 +732,8 @@
   Thaw();
 
   // This for some reason prevents the control from flickering: http://wiki.wxwidgets.org/Flicker-Free_Drawing#No-flickering_for_wxListCtrl_with_wxLC_REPORT_.7C_wxLC_VIRTUAL_style
-  wxIdleEvent idle;
-  wxTheApp->SendIdleEvents(this, idle);
+  // wxIdleEvent idle;
+  // wxTheApp->SendIdleEvents(this, idle);
 }
 
 void RosoutListControl::setSelection(const S_int32& sel)
Index: rxtools/src/rxtools/rxplot_main.py
===================================================================
--- rxtools/src/rxtools/rxplot_main.py	(revision 14716)
+++ rxtools/src/rxtools/rxplot_main.py	(working copy)
@@ -32,7 +32,7 @@
 #
 # Revision $Id$
 
-WXVER = '2.8' 
+WXVER = ['2.8', '2.9']
 import wxversion 
 if wxversion.checkInstalled(WXVER): 
     wxversion.select(WXVER) 
