Index: ogre/Makefile
===================================================================
--- ogre/Makefile	(revision 37811)
+++ ogre/Makefile	(working copy)
@@ -1,12 +1,15 @@
 all: ogre
 
-OGRE_VERSION = ogre_src_v1-7-1
+# OGRE_VERSION = ogre_src_v1-7-1
+# TARBALL = build/$(OGRE_VERSION).tar.bz2
+# TARBALL_URL = http://pr.willowgarage.com/downloads/$(OGRE_VERSION).tar.bz2
+OGRE_VERSION = ogre_src_v1-7-3
 TARBALL = build/$(OGRE_VERSION).tar.bz2
-TARBALL_URL = http://pr.willowgarage.com/downloads/$(OGRE_VERSION).tar.bz2
+TARBALL_URL = http://www.ausparc.com/ros/sources/ogre_src_v1-7-3.tar.bz2
 SOURCE_DIR = build/$(OGRE_VERSION)
 UNPACK_CMD = tar xf
-MD5SUM_FILE = $(OGRE_VERSION).tar.bz2.md5sum
-TARBALL_PATCH = hidden_window.patch hidden_window_glx_fix.patch multi_scene_manager.patch
+# MD5SUM_FILE = $(OGRE_VERSION).tar.bz2.md5sum
+# TARBALL_PATCH = hidden_window.patch hidden_window_glx_fix.patch multi_scene_manager.patch
 #INITIAL_DIR = build/ogre
 include $(shell rospack find mk)/download_unpack_build.mk
 
@@ -22,6 +25,8 @@
 
 CMAKE = cmake
 CMAKE_ARGS = -D CMAKE_INSTALL_PREFIX=$(OGREROOT)\
+						 -D CMAKE_OSX_ARCHITECTURES="x86_64"\
+						 -D OGRE_BUILD_TOOLS="off"\
 						 -D OGRE_BUILD_SAMPLES="off"\
 						 -D OGRE_BUILD_TESTS="off"\
 						 -D CMAKE_BUILD_TYPE="RelWithDebInfo" \
@@ -71,11 +76,19 @@
 	else \
 	  make ogre/lib/pkgconfig/OGRE.pc; \
 	fi
-	@if test ! -f ogre/lib/OGRE/Plugin_CgProgramManager.so; then \
-          echo "BUILD FAILED: Plugin_CgProgramManager.so not created.  Is nvidia-cg-toolkit installed correctly?"; \
-          exit 1; \
-        fi
+	@if test `uname` = Darwin ; then \
+		if test ! -f ogre/lib/Plugin_CgProgramManager.dylib; then \
+			echo "BUILD FAILED: Plugin_CgProgramManager.dylib not created.	 Is nvidia-cg-toolkit installed correctly?"; \
+			exit 1; \
+		fi \
+	else \
+		if test ! -f ogre/lib/OGRE/Plugin_CgProgramManager.so; then \
+			echo "BUILD FAILED: Plugin_CgProgramManager.so not created.	 Is nvidia-cg-toolkit installed correctly?"; \
+			exit 1; \
+		fi \
+	fi
 
+
 ogre/lib/pkgconfig/OGRE.pc: $(SOURCE_DIR)/CMakeLists.txt $(SOURCE_DIR)
         # building OGRE
 	mkdir -p $(ROOT)/$(SOURCE_DIR)/build;
Index: rosdep.yaml
===================================================================
--- rosdep.yaml	(revision 37811)
+++ rosdep.yaml	(working copy)
@@ -1,6 +1,10 @@
 freeimage:
   ubuntu: libfreeimage-dev
   macports: freeimage
+  osx:
+    lion:
+      homebrew:
+        packages: [ freeimage ]
   debian: libfreeimage-dev
   arch: freeimage
   gentoo: freeimage
@@ -13,10 +17,19 @@
     if ! pacman -Q nvidia-cg-toolkit ; then yaourt -S nvidia-cg-toolkit; fi
   gentoo: nvidia-cg-toolkit
   debian: nvidia-cg-toolkit
+  osx:
+    lion:
+      source:
+        uri: 'http://www.ausparc.com/ros/sourcedeps/nvidia-cg.rdmanifest'
+        md5sum: 5877923536b171586bab4e17b25e276f
 libxrandr:
   ubuntu: libxrandr-dev
   debian: libxrandr-dev
   arch: libxrandr
   gentoo: x11-libs/libXrandr
   macports: xorg-libXrandr
+  osx:
+    lion:
+      homebrew:
+        packages: |
   fedora: libXrandr-devel
